activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -68,13 +71,26 @@ def initialize(files, dirs = {}, &block)
     end
 
     def updated?
-      @boot_mutex.synchronize do
-        if @pid != Process.pid
+      if @pid != Process.pid
+        @boot_mutex.synchronize do
```

Comment:
I sneaky-edited which gist my comment was pointing to... just after posting I realised this needs to re-check the conditional inside the lock: `Process.pid` can't have changed, but `@pid` might've been fixed by another thread that did the following while we were waiting for the mutex.